module_utils/kafka_manager.py

```diff
@@ -183,6 +190,63 @@ def delete_topics(self, topics, timeout=None):
                     )
                 )
 
+    def get_coordinator_for_consumer_group(self, consumer_group=None):
+        cluster = self.client.cluster
+        brokers = cluster.brokers()
+
+        # Ensure connections to all brokers
+        for broker in brokers:
+            while not self.client.is_ready(broker.nodeId):
+                self.client.ready(broker.nodeId)
+                self.client.poll()
+
+        # Identify which broker is coordinating this consumer group
+        response = self.send_request_and_get_response(
+                 GroupCoordinatorRequest_v0(consumer_group),
+                 next(iter(brokers)).nodeId
+                 )
+        return response.coordinator_id
+
+    def get_offsets_for_consumer_group(self, consumer_group=None):
+
+        consumer_coordinator = self.get_coordinator_for_consumer_group(
+            consumer_group)
+
+        return self.send_request_and_get_response(
+            OffsetFetchRequest_v2(consumer_group, None), consumer_coordinator)
+
+    def delete_group_offset(self, group_id, topics, check_mode):
+        consumed_offsets = self.get_offsets_for_consumer_group(group_id)
+        consumed_topics = [e[0] for e in consumed_offsets.topics]  # topic name
+
+        changed = False
+
+        topics_partitions = []
+        for topic in topics:
+            topic_name = topic['name']
+            partitons = topic['partitions']
+            if(partitons is None):
+                partitons = [key for key in
+                             self.get_partitions_for_topic(topic_name).keys()]
+            topics_partitions.append((topic_name, partitons))
+            if(topic_name in consumed_topics):
+                changed = True
+
+        consumer_coordinator = self.get_coordinator_for_consumer_group(
```

Comment:
Could you merge your usage of this method with `get_offsets_for_consumer_group`?

Reply:
done